zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/migration/MigrationTaskRegistrationTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.migration;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+
+import com.tngtech.archunit.base.DescribedPredicate;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+
+@AnalyzeClasses(
+    packages = "io.camunda.zeebe.engine.state.migration",
+    importOptions = ImportOption.DoNotIncludeTests.class)
+public class MigrationTaskRegistrationTest {
+
+  @ArchTest()
+  public static final ArchRule MIGRATION_TASKS_MUST_BE_REGISTERED =
+      classes()
+          .that()
+          .implement(MigrationTask.class)
+          .should()
+          .beAssignableFrom(anyOfTheRegisteredMigrationTasks());
```

Comment:
:thought_balloon: This reads a bit weird, I would have expected something like `.isOneOf(registeredMigrationsTaskClasses())` but as long as it works...

Reply:
That would read nicer, but this is the ArchUnit API so not much I can do about it 🤷